addons/im_livechat/static/src/embed/common/chatbot/chatbot_model.js

```diff
@@ -59,10 +59,10 @@ export class Chatbot extends Record {
         if (!this.currentStep || this.currentStep.completed || !this.thread) {
             return;
         }
-        const { Message: messages = [] } = this.store.insert(this.currentStep.data, { html: true });
-        this.currentStep.message =
-            messages[0] ??
-            this.store.Message.insert(
+        if (!this.currentStep.message) {
+            // Thread is not persisted thus messages do not exist on the server,
+            // create them now on the client side.
+            this.currentStep.message = this.store.Message.insert(
```

Comment:
Slightly out-of-scope of PR, but I think the insert of message could be more declarative by setting only `{ id: temporaryId,  chatbotStep: this }`, the rest of data looks deduced by `chatbotStep`.